src/Middleware/HttpOverrides/src/ForwardedHeadersMiddleware.cs

```diff
@@ -167,14 +167,31 @@ public void ApplyForwarders(HttpContext context)
             entryCount = Math.Max(forwardedHost.Length, entryCount);
         }
 
+        if (_options.ForwardedHeaders.HasFlag(ForwardedHeaders.XForwardedPort))
+        {
+            checkPort = true;
+            forwardedPort = requestHeaders.GetCommaSeparatedValues(_options.ForwardedPortHeaderName);
+            if (_options.RequireHeaderSymmetry
+                && ((checkFor && forwardedFor!.Length != forwardedPort.Length)
+                    || (checkProto && forwardedProto!.Length != forwardedPort.Length)
+                    || (checkHost && forwardedHost!.Length != forwardedPort.Length)
+                    || (checkPort && forwardedPort!.Length != forwardedPort.Length)))
+            {
+                _logger.LogWarning(1, "Parameter count mismatch between X-Forwarded-Prefix and X-Forwarded-Host and X-Forwarded-Port and X-Forwarded-For or X-Forwarded-Proto.");
+                return;
+            }
+            entryCount = Math.Max(forwardedPort.Length, entryCount);
+        }
+
         if (_options.ForwardedHeaders.HasFlag(ForwardedHeaders.XForwardedPrefix))
         {
             checkPrefix = true;
             forwardedPrefix = requestHeaders.GetCommaSeparatedValues(_options.ForwardedPrefixHeaderName);
             if (_options.RequireHeaderSymmetry
                 && ((checkFor && forwardedFor!.Length != forwardedPrefix.Length)
                     || (checkProto && forwardedProto!.Length != forwardedPrefix.Length)
-                    || (checkHost && forwardedHost!.Length != forwardedPrefix.Length)))
+                    || (checkHost && forwardedHost!.Length != forwardedPrefix.Length)
+                    || (checkPort && forwardedPort!.Length != forwardedPrefix.Length)))
             {
                 _logger.LogWarning(1, "Parameter count mismatch between X-Forwarded-Prefix and X-Forwarded-Host and X-Forwarded-For or X-Forwarded-Proto.");
```

Comment:
Seems like the log message probably needs to be updated.

Reply:
```suggestion
                _logger.LogWarning(1, "Parameter count mismatch between X-Forwarded-Prefix and X-Forwarded-Host and X-Forwarded-For and X-Forwarded-Proto or X-Forwarded-Port.");
```